### DISCUSSION TITLE
有没有可能把milkdown当做普通的富文本编辑器使用呢？

### DISCUSSION THREAD

**Discussion by ununian** [Milkdown/milkdown#536] [Q&A] [answered]
我找了一圈富文本编辑器，发现 milkdown 的功能还有代码设计都比较符合我的需求，除了一点，就是我们需要对行内的文本做标记，例如文字颜色和下划线的颜色。我知道这是 markdown 的局限性，那么我如果直接保存 prosemirror 的数据结构是不是就没这种限制了？如果要实现这种需求，改动的难度大吗？
或者我换另一种思路，有没有办法在 tiptap 里面使用 milkdown 的那些插件呢，兼容的成本会不会很高？

**Comment by Saul-Mirone** [comment, marked as answer]
是有可能的，你如果不打算用markdown，可以直接用prosemirror的结构，要实现起来就和普通的prosemirror mark差不多，不会很复杂。但可能会影响一些依赖parser和serializer的功能，所以还是推荐扩展markdown，按照自定义语法来实现。

tiptap里面用milkdown的插件的兼容成本估计会比较高，因为tiptap有自己的封装层。
